extensions/2.0/Khronos/KHR_lights_environment/README.md

```diff
@@ -0,0 +1,143 @@
+Image-based lights consist of an environment map that represents specular radiance for the scene as well as irradiance information.  
+This can be used on it's own - ie a glTF asset with only environment map data - for a usecase where the IBL needs to be distributed.  
+It can also be used together with model data, for usecases where a model shall be displayed in a defined environment.  
+
+Many 3D tools and engines support image-based global illumination but the exact technique and data formats employed vary.  
+Using this extension, tools can export and engines can import image-based lights and the result should be highly consistent.   
+
+This extension specifies exactly one way to format and reference the environment map to be used.  
+The goals of this are two-fold.  
+First, it makes implementing support for this extension easier.  
+Secondly, it ensures that rendering of the image-based lighting is consistent across runtimes.
+
+A conforming implementation of this extension must be able to load the image-based environment data and render the PBR materials using this lighting.  
+
+This extension is based on EXT_lights_image_based, with added support for KTX2 as a carrier for [cubemap and] HDR texture formats.  
+
+The environment light is defined by a cubemap.  
+[TBD - shall a separate KHR_texture_cubemap extension be created?]
+Cubemaps can be supplied pre-filtered, by including the pre-filtered mip-levels, otherwise filtering will be performed at load time.  
+[TBD - point to relevant filtering algorithm]  
+If a compressed texture format is used then pre-filtered mip-levels must be included.  
+
+## Declaring an environment light
+
+The KHR_lights_environment extension defines an array of image-based lights at the root of the glTF and then each scene can reference one.  
+Each environment light definition consists of a single cubemap that describes the specular radiance of the scene, the l=2 spherical harmonics coefficients for diffuse irradiance and intensity values.  
+The cubemap is defined by texture references to a KTX2 file containing a cubemap, these files can contain compressed textures using KHR_texture_basisu  
+or use a float texture format as defined by KHR_texture_float_bc6h or KHR_texture_float_rgb95e.   
+
+
+```json
+"extensions": {
+    "KHR_lights_environment" : {
+        "lights": [
+            {
+                "intensity": 1.0,
+                "irradianceCoefficients": [...3 x 9 array of floats...],
+                "specularCubemaps": [... Reference to KTX2 cubemaps...],
+            }
+        ]
+    }
+}
+```
+
+## Specular Radiance Cubemaps
+
+The cubemap used for specular radiance is defined as a cubemap containing separate images for each cube face. 
+The mip levels shall evenly map to roughness values from 0 to 1 in the PBR material and should be generated with a principled multi-scatter GGX normal distribution. The data in the maps represents illuminance in candela per square meter (nit).
```

Comment:
Do we need to specify the cube map mipmaps as opposed to just the top level one?  This is again an optimization because they are derived data.  Only the top most cube map is required.  Also by providing them we are deciding that they can only implement roughness via using a series of cube-maps, when in reality there are alternative implementations.  For example, one could use a method that retain more resolution than a mip chain even at high roughness levels.  Or one could have create a tree where one specifies LOD and roughness.  For example, at a 0 roughness level one could still have a mip chain for that cube map so that it avoids jaggies when zoomed out.

I would much prefer we focus on non-duplicated non-derived data in the specification of this extension.  If someone prefers to send duplicate pre-computed data, then should specify a vendor extension for that payload IMO.

Reply:
> * Unity: Sky Box

I can see that the first three could be potential naming kandidates - however, I feel that the skybox is not what we are defining.  
It's more like a leftover from the 90's when you wrapped your game with a view of the sky.
According to Unity manual there is no light contribution from skybox:
https://docs.unity3d.com/Manual/skyboxes-using.html#:~:text=%20To%20make%20Unity%20do%20this%3A%20%201,and%2C%20from%20the%20list%2C%20click%20Skybox.%20More%20

It seems to me that `environment` is used more in academia, for instance one of the early papers:
https://graphics.stanford.edu/papers/ravir_thesis/chapter4.pdf
As such I like it just because it's not attached to an engine implementation - it's more of a description of the methodology.

Reply:
> Do we need to specify the cube map mipmaps as opposed to just the top level one?

I need to work on the wording - my intention is to allow specification of the roughness mipmap in the file - for instance if you use a compressed format you would generally not want to filter then compress at runtime :-)

For uncompressed formats I think it makes sense to allow exclusion of the (roughness) mipmaps - generating this on the fly is both quite simple and quick.

Reply:
> According to Unity manual there is no light contribution from skybox:

Unity does take lighting contributions from a Skybox: https://learn.unity.com/tutorial/creating-a-high-dynamic-range-image-skybox#5f358fb3edbc2a1563767802

> I need to work on the wording - my intention is to allow specification of the roughness mipmap in the file - for instance if you use a compressed format you would generally not want to filter then compress at runtime :-)

I believe you are presupposing implementation details which are not necessary in this mipmap chain, rather than specifying the physical inputs which are more universal.

The last time this extension was attempted, it was moved to EXT in part because it contained pre-filtered MIPS.  Here is another Three.js contributor making that case: https://twitter.com/EmmettLalish/status/1367196117801181184. And here is the author of Filmant making that case: https://twitter.com/romainguy/status/1367236298218426369.  (And remember I am also making this case.)

Reply:
While I agree having to provide the full mip chain complicates the extension and in turn any tooling wishing to create assets using the extension, I don't see a good way around that for compressed formats without the runtime decompress the textures (and keep them decompressed when uploading to the GPU). 

Making it optional for uncompressed formats seems reasonable enough, though its trading tooling implementation complexity for runtime implementation complexity. I can see arguments either way there.

Can you elaborate on what you meant re:
> I believe you are presupposing implementation details which are not necessary in this mipmap chain, rather than specifying the physical inputs which are more universal.